include/networkit/algebraic/CSRGeneralMatrix.hpp

```diff
@@ -0,0 +1,1226 @@
+        this->nonZeros = std::vector<double>(nnz);
+
+#pragma omp parallel for
+        for (omp_index i = 0; i < static_cast<omp_index>(nRows); ++i) {
+            for (index k = 0; k < columnIdx[i].size(); ++k) {
+                this->columnIdx[rowIdx[i] + k] = columnIdx[i][k];
+                nonZeros[rowIdx[i] + k] = values[i][k];
+            }
+        }
+    }
+
+    /**
+     * Constructs the @a nRows x @a nCols Matrix from the elements at position @a
+     * positions with values @values.
+     * @param nRows Defines how many rows this matrix has.
+     * @param nCols Defines how many columns this matrix has.
+     * @param rowIdx The rowIdx vector of the CSR format.
+     * @param columnIdx The columnIdx vector of the CSR format.
+     * @param nonZeros The nonZero vector of the CSR format. Should be as long as
+     * the @a columnIdx vector.
+     * @param zero The zero element (default is 0).
+     * @param isSorted True, if the triplets are sorted per row. Default is false.
+     */
+    CSRGeneralMatrix(count nRows, count nCols, const std::vector<index> &rowIdx,
+                     const std::vector<index> &columnIdx, const std::vector<ValueType> &nonZeros,
+                     ValueType zero = 0, bool isSorted = false)
+        : rowIdx(rowIdx), columnIdx(columnIdx), nonZeros(nonZeros), nRows(nRows), nCols(nCols),
+          isSorted(isSorted), zero(zero) {}
+
+    /** Default copy constructor */
+    CSRGeneralMatrix(const CSRGeneralMatrix &other) = default;
+
+    /** Default move constructor */
+    CSRGeneralMatrix(CSRGeneralMatrix &&other) noexcept = default;
+
+    /** Default destructor */
+    virtual ~CSRGeneralMatrix() = default;
+
+    /** Default move assignment operator */
+    CSRGeneralMatrix &operator=(CSRGeneralMatrix &&other) noexcept = default;
+
+    /** Default copy assignment operator */
+    CSRGeneralMatrix &operator=(const CSRGeneralMatrix &other) = default;
+
+    /**
+     * Compares this matrix to @a other and returns true if the shape and zero
+     * element are the same as well as
+     * all entries, otherwise returns false.
+     * @param other
+     */
+    bool operator==(const CSRGeneralMatrix &other) const {
+        bool equal = nRows == other.nRows && nCols == other.nCols && zero == other.zero;
+        if (equal)
+            forNonZeroElementsInRowOrder([&](index i, index j, ValueType value) {
+                if (other(i, j) != value) {
+                    equal = false;
+                    return;
+                }
+            });
+
+        return equal;
+    }
+
+    /**
+     * Compares this matrix to @a other and returns false if the shape and zero
+     * element are the same as well as
+     * all entries, otherwise returns true.
+     * @param other
+     */
+    bool operator!=(const CSRGeneralMatrix &other) const { return !((*this) == other); }
+
+    /**
+     * @return Number of rows.
+     */
+    inline count numberOfRows() const noexcept { return nRows; }
```

Comment:
`inline` is redundant if a method is defined within `class { ... }` or `struct { ... }`.